common/v2/features/SendAssets/components/fields/AssetDropdown.tsx

```diff
@@ -24,21 +33,23 @@ class AssetOption extends React.PureComponent<OptionComponentProps> {
 }
 
 function AssetDropdown({ assets, name, value, onSelect }: Props<Asset>) {
-  const filteredAssets: Asset[] = assets.filter(
-    (asset, index) => assets.map(assetObj => assetObj.uuid).indexOf(asset.uuid) >= index
-  ); /* Removes duplicates */
+  const filteredAssets: Asset[] = assets
+    .filter((asset, index) => assets.map(assetObj => assetObj.uuid).indexOf(asset.uuid) >= index)
+    .map(asset => ({ label: asset.name, id: asset.uuid, ...asset })); /* Removes duplicates */
```

Comment:
Since there are performance issue let's trying setting the state only once inside a `useEffect`.